scipy/spatial/ckdtree/src/partial_sort.h

```diff
@@ -64,8 +64,11 @@ static int partition_node_indices(const double *data,
      *    modified as noted above.
      */
 
-    IndexComparator index_comparator = {.data=data, .split_dim=split_dim, 
-                                        .n_dims = n_dims};
+    IndexComparator index_comparator;
```

Comment:
This is correct C, but we are using C++. We can use a constructor for initalization.